deepnote_toolkit/kernel_checkpoint/checkpoint.py

```
@@ -0,0 +1,127 @@
+
+
+def try_restore_checkpoint(store: SnapshotStore, key: str) -> tuple[dict[str, Any], RestoreReport] | None:
+    """Attempt to restore from `store`. Returns (globals, report) or None if absent.
+
+    Callers MERGE the returned dict into their target namespace — this module
+    intentionally doesn't mutate any caller state.
+    """
+    payload = store.read(key)
+    if payload is None:
+        return None
+    restored: dict[str, Any] = dill.loads(payload)
```

Ruff (0.15.14)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)